File: examples-sdk/java/030-basic-trilane-messaging/Main.java
// 030-basic-trilane-messaging — Java SDK equivalent
// Compile: vil compile --from java --input 030-basic-trilane-messaging/Main.java --release

import dev.vil.*;

public class Main {
    public static void main(String[] args) {
        VilServer server = new VilServer("ecommerce-order-pipeline", 8080);
        ServiceProcess gateway = new ServiceProcess("gateway");
        server.service(gateway);
        ServiceProcess fulfillment = new ServiceProcess("fulfillment");
        fulfillment.endpoint("GET", "/status", "fulfillment_status");
        server.service(fulfillment);
        server.compile();
    }
}

File: examples-sdk/csharp/030-basic-trilane-messaging/Main.cs
// 030-basic-trilane-messaging — C# SDK equivalent
// Compile: vil compile --from csharp --input 030-basic-trilane-messaging/Main.cs --release

#r "sdk/csharp/Vil.cs"

var server = new VilServer("ecommerce-order-pipeline", 8080);
var gateway = new ServiceProcess("gateway");
server.Service(gateway);
var fulfillment = new ServiceProcess("fulfillment");
fulfillment.Endpoint("GET", "/status", "fulfillment_status");
server.Service(fulfillment);
server.Compile();
